src/shared/creds-util.c

```diff
@@ -660,16 +660,15 @@ int encrypt_credential_and_warn(
 #if HAVE_TPM2
         bool try_tpm2;
         if (sd_id128_equal(with_key, _CRED_AUTO)) {
```

Comment:
as mentioned, change this to `if (sd_id128_in_set(with_key, _CRED_AUTO, _CRED_AUTO_INITRD)) {` 

and then drop the _CRED_AUTO_INITRD branch

Reply:
Yes, I am doing it in a different commit.

Reply:
Should be done in the new commit, and the log_warning message as adapted.